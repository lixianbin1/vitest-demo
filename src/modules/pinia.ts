import { App } from 'vue'
import { createPinia } from 'pinia'

export const PiniaStore = createPinia()
export const install = (app:App) => {
  app.use(PiniaStore)
}
export default install