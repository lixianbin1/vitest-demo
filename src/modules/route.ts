import { App } from 'vue'
import { createRouter,createWebHashHistory } from 'vue-router'
import { setupLayouts } from 'virtual:generated-layouts'
import generatedRoutes from '~pages'

export const routes = setupLayouts(generatedRoutes)
export const router = createRouter({
  history: createWebHashHistory(),
  routes,
})
export const install = (app:App) => {
  app.use(router)
}
export default install