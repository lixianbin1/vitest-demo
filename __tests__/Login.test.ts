import {expect,test,vi,it,describe,bench} from 'vitest'
import { mount } from '@vue/test-utils'
import {render} from '@testing-library/vue'
import LoginPage from '../src/pages/login/index.vue'
import {i18n} from '../src/modules/i18n'
import {router} from '../src/modules/route'

describe('Login.vue', () => {
  test('renders login form', () => {
    const wrapper = mount(LoginPage,{
      global: {
        plugins:[i18n,router]
      }
    })

    // 检查是否渲染了正确的元素
    expect(wrapper.find('#login').exists()).toBe(true)
    expect(wrapper.find('.header').exists()).toBe(true)
    expect(wrapper.find('.main').exists()).toBe(true)
    expect(wrapper.find('.login_box').exists()).toBe(true)
    expect(wrapper.find('.submit-btn').exists()).toBe(true)
  },{time:1000})

  it('renders correct default data', async () => {
    const wrapper = mount(LoginPage,{
      global: {
        plugins:[i18n,router]
      }
    })

    // 检查默认数据是否正确
    expect(wrapper.vm.state.form.ptCode).toBe('FF038')
    expect(wrapper.vm.state.form.username).toBe('Admin')
    expect(wrapper.vm.state.form.password).toBe('123456')
  })

  // 更多的测试...
})

test('Login页面的快照',()=>{
  const { container } = render(LoginPage,{
    global: {
      plugins:[i18n,router]
    }
  })
  expect(container).toMatchSnapshot()
})

// test('函数值快照', () => {
//   const result = toUpperCase('foobar')
//   expect(result).toMatchSnapshot()
// })