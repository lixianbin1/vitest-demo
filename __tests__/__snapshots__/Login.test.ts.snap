// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login页面的快照 1`] = `
<div>
  <div
    data-v-45258083=""
    id="login"
  >
    <header
      class="el-header header"
      data-v-45258083=""
      flex=""
      items-center=""
    >
      
      <h1
        class="title"
        data-v-45258083=""
        flex-1=""
        font-800=""
        text-18px=""
      >
        Welcome to the template vue3
      </h1>
      <div
        class="lang"
        data-v-45258083=""
        flex=""
        flex-1=""
        justify-end=""
      >
        <button
          aria-disabled="false"
          class="el-button is-link userinfo"
          data-v-45258083=""
          data-v-6f72e247=""
          type="button"
        >
          <!--v-if-->
          <span
            class=""
          >
            
            <i
              class="el-icon"
              data-v-6f72e247=""
            >
              
              <svg
                data-v-6f72e247=""
                height="1.2em"
                viewBox="0 0 1024 1024"
                width="1.2em"
              >
                <path
                  d="M640 384v64H448a128 128 0 0 0-128 128v128a128 128 0 0 0 128 128h320a128 128 0 0 0 128-128V576a128 128 0 0 0-64-110.848V394.88c74.56 26.368 128 97.472 128 181.056v128a192 192 0 0 1-192 192H448a192 192 0 0 1-192-192V576a192 192 0 0 1 192-192z"
                  fill="currentColor"
                />
                <path
                  d="M384 640v-64h192a128 128 0 0 0 128-128V320a128 128 0 0 0-128-128H256a128 128 0 0 0-128 128v128a128 128 0 0 0 64 110.848v70.272A192.06 192.06 0 0 1 64 448V320a192 192 0 0 1 192-192h320a192 192 0 0 1 192 192v128a192 192 0 0 1-192 192z"
                  fill="currentColor"
                />
              </svg>
              
            </i>
            
          </span>
        </button>
      </div>
      
    </header>
    <main
      class="el-main main"
      data-v-45258083=""
    >
      
      <div
        class="login_box"
        data-v-45258083=""
      >
        <form
          class="el-form el-form--default el-form--label-right"
          data-v-45258083=""
        >
          
          <div
            class="el-form-item asterisk-left"
            data-v-45258083=""
          >
            
            
            <!--v-if-->
            
            
            <div
              class="el-form-item__content"
            >
              
              <h2
                class="title-login"
                data-v-45258083=""
                text-20px=""
              >
                Account
              </h2>
              
              <transition-group-stub
                appear="false"
                css="true"
                name="el-zoom-in-top"
                persisted="false"
              >
                <!--v-if-->
              </transition-group-stub>
            </div>
          </div>
          <div
            class="el-form-item is-required asterisk-left"
            data-v-45258083=""
          >
            
            
            <!--v-if-->
            
            
            <div
              class="el-form-item__content"
            >
              
              <span
                data-v-45258083=""
              >
                user name
              </span>
              <div
                class="el-input el-input--prefix el-input--suffix"
                data-v-45258083=""
              >
                <!-- input -->
                
                <!-- prepend slot -->
                <!--v-if-->
                <div
                  class="el-input__wrapper"
                >
                  <!-- prefix slot -->
                  <span
                    class="el-input__prefix"
                  >
                    <span
                      class="el-input__prefix-inner"
                    >
                      
                      <i
                        class="el-icon"
                        data-v-45258083=""
                        i-carbon:user-avatar=""
                      >
                        
                        
                      </i>
                      
                      <!--v-if-->
                    </span>
                  </span>
                  <input
                    autocomplete="off"
                    class="el-input__inner"
                    tabindex="0"
                    type="text"
                  />
                  <!-- suffix slot -->
                  <!--v-if-->
                </div>
                <!-- append slot -->
                <!--v-if-->
                
              </div>
              
              <transition-group-stub
                appear="false"
                css="true"
                name="el-zoom-in-top"
                persisted="false"
              >
                <!--v-if-->
              </transition-group-stub>
            </div>
          </div>
          <div
            class="el-form-item is-required asterisk-left"
            data-v-45258083=""
          >
            
            
            <!--v-if-->
            
            
            <div
              class="el-form-item__content"
            >
              
              <span
                data-v-45258083=""
              >
                Userid
              </span>
              <div
                class="el-input el-input--prefix el-input--suffix"
                data-v-45258083=""
              >
                <!-- input -->
                
                <!-- prepend slot -->
                <!--v-if-->
                <div
                  class="el-input__wrapper"
                >
                  <!-- prefix slot -->
                  <span
                    class="el-input__prefix"
                  >
                    <span
                      class="el-input__prefix-inner"
                    >
                      
                      <i
                        class="el-icon"
                        data-v-45258083=""
                        i-carbon:user-avatar=""
                      >
                        
                        
                      </i>
                      
                      <!--v-if-->
                    </span>
                  </span>
                  <input
                    autocomplete="off"
                    class="el-input__inner"
                    tabindex="0"
                    type="text"
                  />
                  <!-- suffix slot -->
                  <!--v-if-->
                </div>
                <!-- append slot -->
                <!--v-if-->
                
              </div>
              
              <transition-group-stub
                appear="false"
                css="true"
                name="el-zoom-in-top"
                persisted="false"
              >
                <!--v-if-->
              </transition-group-stub>
            </div>
          </div>
          <div
            class="el-form-item asterisk-left"
            data-v-45258083=""
          >
            
            
            <!--v-if-->
            
            
            <div
              class="el-form-item__content"
            >
              
              <span
                data-v-45258083=""
              >
                Password
              </span>
              <div
                class="el-input el-input--prefix el-input--suffix"
                data-v-45258083=""
              >
                <!-- input -->
                
                <!-- prepend slot -->
                <!--v-if-->
                <div
                  class="el-input__wrapper"
                >
                  <!-- prefix slot -->
                  <span
                    class="el-input__prefix"
                  >
                    <span
                      class="el-input__prefix-inner"
                    >
                      
                      <i
                        class="el-icon"
                        data-v-45258083=""
                        i-carbon:password=""
                      >
                        
                        
                      </i>
                      
                      <!--v-if-->
                    </span>
                  </span>
                  <input
                    autocomplete="off"
                    class="el-input__inner"
                    tabindex="0"
                    type="password"
                  />
                  <!-- suffix slot -->
                  <span
                    class="el-input__suffix"
                  >
                    <span
                      class="el-input__suffix-inner"
                    >
                      
                      
                      
                      <!--v-if-->
                      
                      <!--v-if-->
                      <i
                        class="el-icon el-input__icon el-input__password"
                      >
                        
                        <svg
                          viewBox="0 0 1024 1024"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M876.8 156.8c0-9.6-3.2-16-9.6-22.4-6.4-6.4-12.8-9.6-22.4-9.6-9.6 0-16 3.2-22.4 9.6L736 220.8c-64-32-137.6-51.2-224-60.8-160 16-288 73.6-377.6 176C44.8 438.4 0 496 0 512s48 73.6 134.4 176c22.4 25.6 44.8 48 73.6 67.2l-86.4 89.6c-6.4 6.4-9.6 12.8-9.6 22.4 0 9.6 3.2 16 9.6 22.4 6.4 6.4 12.8 9.6 22.4 9.6 9.6 0 16-3.2 22.4-9.6l704-710.4c3.2-6.4 6.4-12.8 6.4-22.4Zm-646.4 528c-76.8-70.4-128-128-153.6-172.8 28.8-48 80-105.6 153.6-172.8C304 272 400 230.4 512 224c64 3.2 124.8 19.2 176 44.8l-54.4 54.4C598.4 300.8 560 288 512 288c-64 0-115.2 22.4-160 64s-64 96-64 160c0 48 12.8 89.6 35.2 124.8L256 707.2c-9.6-6.4-19.2-16-25.6-22.4Zm140.8-96c-12.8-22.4-19.2-48-19.2-76.8 0-44.8 16-83.2 48-112 32-28.8 67.2-48 112-48 28.8 0 54.4 6.4 73.6 19.2zM889.599 336c-12.8-16-28.8-28.8-41.6-41.6l-48 48c73.6 67.2 124.8 124.8 150.4 169.6-28.8 48-80 105.6-153.6 172.8-73.6 67.2-172.8 108.8-284.8 115.2-51.2-3.2-99.2-12.8-140.8-28.8l-48 48c57.6 22.4 118.4 38.4 188.8 44.8 160-16 288-73.6 377.6-176C979.199 585.6 1024 528 1024 512s-48.001-73.6-134.401-176Z"
                            fill="currentColor"
                          />
                          <path
                            d="M511.998 672c-12.8 0-25.6-3.2-38.4-6.4l-51.2 51.2c28.8 12.8 57.6 19.2 89.6 19.2 64 0 115.2-22.4 160-64 41.6-41.6 64-96 64-160 0-32-6.4-64-19.2-89.6l-51.2 51.2c3.2 12.8 6.4 25.6 6.4 38.4 0 44.8-16 83.2-48 112-32 28.8-67.2 48-112 48Z"
                            fill="currentColor"
                          />
                        </svg>
                        
                      </i>
                      <!--v-if-->
                      <!--v-if-->
                    </span>
                  </span>
                </div>
                <!-- append slot -->
                <!--v-if-->
                
              </div>
              
              <transition-group-stub
                appear="false"
                css="true"
                name="el-zoom-in-top"
                persisted="false"
              >
                <!--v-if-->
              </transition-group-stub>
            </div>
          </div>
          <div
            class="el-form-item asterisk-left"
            data-v-45258083=""
            style="margin-top: 30px;"
          >
            
            
            <!--v-if-->
            
            
            <div
              class="el-form-item__content"
            >
              
              <div
                data-v-45258083=""
                style="text-align: center; width: 100%;"
              >
                <button
                  aria-disabled="false"
                  class="el-button el-button--primary is-plain submit-btn bthColor"
                  data-v-45258083=""
                  type="button"
                >
                  <!--v-if-->
                  <span
                    class=""
                  >
                    
                    Login
                    
                  </span>
                </button>
              </div>
              
              <transition-group-stub
                appear="false"
                css="true"
                name="el-zoom-in-top"
                persisted="false"
              >
                <!--v-if-->
              </transition-group-stub>
            </div>
          </div>
          <div
            class="el-form-item asterisk-left"
            data-v-45258083=""
          >
            
            
            <!--v-if-->
            
            
            <div
              class="el-form-item__content"
            >
              
              <div
                class="footerTitle footerTitleTwo"
                data-v-45258083=""
              >
                <p
                  data-v-45258083=""
                >
                  v2024.03.25
                </p>
              </div>
              
              <transition-group-stub
                appear="false"
                css="true"
                name="el-zoom-in-top"
                persisted="false"
              >
                <!--v-if-->
              </transition-group-stub>
            </div>
          </div>
          
        </form>
      </div>
      
    </main>
    <footer
      class="el-footer"
      data-v-45258083=""
      style="align-items: center; display: flex;"
    >
      
      <span
        class="footerTitle"
        data-v-45258083=""
      >
        © the Industry Associations XXXXXX. All rights reserved.
      </span>
      
    </footer>
  </div>
</div>
`;
